[Doc]: update minimum gcc version?

### Documentation Link

https://matplotlib.org/devdocs/devel/dependencies.html#c-compiler

### Problem

The minimum stated version of gcc for building matplotlib is currently 4.8.1.  When trying to build matplotlib with (our system default) gcc v4.8.5, I get

```
/usr/include/c++/4.8.2/bits/c++0x_warning.h:32:2: error: #error This file requires compiler and library support for the ISO C++ 2011 standard. This support is currently experimental, and must be enabled with the -std=c++11 or -std=gnu++11 compiler options.
```

On the same system, if I point at gcc v8.1.0, the build works fine.

### Suggested improvement

I suspect the simplest option would be to increase the stated minimum version in the installation guide, though I do not know what version that would be.  I note that 4.8.5 came out in 2015.

--- a/doc/devel/dependencies.rst
+++ b/doc/devel/dependencies.rst
@@ -213,7 +213,8 @@ C++ compiler
 
 Matplotlib requires a C++ compiler that supports C++11.
 
-- `gcc 4.8.1 <https://gcc.gnu.org/projects/cxx-status.html#cxx11>`_ or higher
+- `gcc 4.8.1 <https://gcc.gnu.org/projects/cxx-status.html#cxx11>`_ or higher.  For gcc <6.5 you will
+  need to set ``$CFLAGS=-std=c++11`` to enable C++11 support.
 - `clang 3.3 <https://clang.llvm.org/cxx_status.html>`_ or higher
 - `Visual Studio 2015
   <https://docs.microsoft.com/en-us/cpp/overview/visual-cpp-language-conformance?view=msvc-140>`_

